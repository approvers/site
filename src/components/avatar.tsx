import styles from "../scss/components/avatar.module.scss";
import { useImage } from "react-image";

const ALTERNATIVE = "/alternative.png";

export const Avatar = ({ name, src }: { name: string; src?: string }): JSX.Element => {
  const { src: guarded } = useImage({ srcList: [src ?? "", ALTERNATIVE], useSuspense: false });
  return <img src={guarded ?? src} alt={name} className={styles.avatar} />;
};
