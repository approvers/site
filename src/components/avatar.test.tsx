import { Avatar } from "./avatar";
import React from "react";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer
    .create(<Avatar name="MikuroXina" src="https://github.com/MikuroXina.png" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
