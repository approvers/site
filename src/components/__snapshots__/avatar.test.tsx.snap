// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<img
  alt="MikuroXina"
  className="avatar"
  src="https://github.com/MikuroXina.png"
/>
`;
